tests/reg_issue5352/reg_issue5352.fz

```diff
@@ -0,0 +1,25 @@
+# This file is part of the Fuzion language implementation.
+#
+# The Fuzion language implementation is free software: you can redistribute it
+# and/or modify it under the terms of the GNU General Public License as published
+# by the Free Software Foundation, version 3 of the License.
+#
+# The Fuzion language implementation is distributed in the hope that it will be
+# useful, but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public
+# License for more details.
+#
+# You should have received a copy of the GNU General Public License along with The
+# Fuzion language implementation.  If not, see <https://www.gnu.org/licenses/>.
+
+
+# -----------------------------------------------------------------------
+#
+#  Tokiwa Software GmbH, Germany
+#
+#  Source code of Fuzion test reg_issue5352
+#
+# -----------------------------------------------------------------------
+
+q(_ ()->unit) =>
+q ({_:=1})
```

Comment:
can you add test cases like these:
```suggestion
q ({_:=1})
q ({_:=say "hi"})
q ({_:=f})
f => say "hi"
```
and make sure that `say` is actually called.